refactor: improve accessibility for theme menu checkboxes and remove redundant aria-labels

ledjay opened the pull request:
- Removed unused imports and variables from NeedItem component
- Simplified NeedItem by removing complex aria-label generation logic
- Added aria-label directly to checkbox native inputs in Needs and ResultsSection components
- Moved aria-label from wrapper div to button element in ThemeItem
- Added aria-hidden to count displays to prevent screen reader duplication
- Removed redundant theme title display in Result

ledjay committed:
refactor: improve accessibility for theme menu checkboxes and remove redundant aria-labels

- Removed unused imports and variables from NeedItem component
- Simplified NeedItem by removing complex aria-label generation logic
- Added aria-label directly to checkbox native inputs in Needs and ResultsSection components
- Moved aria-label from wrapper div to button element in ThemeItem
- Added aria-hidden to count displays to prevent screen reader duplication
- Removed redundant theme title display in Result

kaaloo commented on apps/client/src/components/Pages/recherche/ThemeMenu/Needs.tsx:
```diff
@@ -123,7 +123,7 @@ const Needs = React.forwardRef<HTMLDivElement | null, {}>((props, ref) => {
         <Checkbox
           legend={t("Recherche.theme", "Thème")}
-          className="[&_.fr-checkbox-group:first-child]:border-default-grey [&_.fr-checkbox-group:first-child]:border-b [&_legend]:sr-only"
+          className="[&_.fr-checkbox-group:first-child]:border-default-grey [&_.fr-checkbox-group:first-child]:border-b [&_legend]:sr-only [&_legend]:hidden"
```
Wdyt? @ledjay

ledjay replied:
That's actually a good point but in this context we just need to remove the legent altogether to avoid vocalization overload :)
In short, I removed the legend ^^

kaaloo commented on apps/client/src/components/Pages/recherche/ThemeMenu/Needs.tsx:
```diff
@@ -143,8 +145,10 @@ const Needs = React.forwardRef<HTMLDivElement | null, {}>((props, ref) => {
+                  "checked": selected,
+                  "onChange": () => selectNeed(need._id),
+                  "className": "!border",
+                  "aria-label": `${need[locale]?.text || ""} ${nbDispositifsByNeed[need._id.toString()] || 0} ${t("Recherche.fiches", "fiches")})`,
```
Seems accurate @ledjay

kaaloo commented on apps/client/src/components/Pages/recherche/ThemeMenu/ResultsSection.tsx:
```diff
@@ -31,6 +31,13 @@ const ResultsSection: React.FC<Props> = ({ theme, needs }) => {
+  needs.forEach((need) => {
+    const results = queryDispositifs({ ...query, needs: [need._id], themes: [] }, dispositifs, allNeeds);
+    nbDispositifsByNeed[need._id.toString()] = results.matches.length;
+  });
```
Also seems valid @ledjay .

ledjay replied:
Woops, my bad it was way overkill.
Just a little tweak and them *poof* it's clean again :)

kaaloo reviewed (changes requested):
Looks like Gemini has some valid suggestions...

ledjay committed:
refactor: optimize need count calculation by using shared context

- Removed duplicate nbDispositifsByNeed calculation from ResultsSection component
- Added ThemeMenuContext import and useContext hook to access shared need counts
- Fixed extra closing parenthesis in aria-label for need checkboxes
- Removed unused legend prop and associated CSS classes for hiding legend in Needs component

ledjay commented:
@kaaloo I made a few tweaks to implement the gemini suggestions, LGTY ?

kaaloo reviewed (changes requested):
There's a styling change in that the headers here are not in bold.

<img width="1235" height="771" alt="Image" src="https://github.com/user-attachments/assets/cf7354be-9fe3-400a-b838-8ac88f437a74" />

ledjay committed:
refactor: improve styling for needs fieldset legend in ResultsSection

- Added custom font styling for fieldset legend using CSS layer
- Added padding utilities for legend element positioning
- Removed margin from fieldset and div elements for better layout control

ledjay commented:
> There's a styling change in that the headers here are not in bold.
> 
> <img alt="Image" width="1235" height="771" src="https://private-user-images.githubusercontent.com/67010/518194203-cf7354be-9fe3-400a-b838-8ac88f437a74.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQxNzMyODIsIm5iZiI6MTc2NDE3Mjk4MiwicGF0aCI6Ii82NzAxMC81MTgxOTQyMDMtY2Y3MzU0YmUtOWZlMy00MDBhLWI4MzgtOGFjODhmNDM3YTc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTI2VDE2MDMwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI2ZWYzNWVjNThmNzczOGVhYjdkMDFiZjUyNmRmOGY4Zjc1MjMxODYyZTMyMTlkZDIwNTM3OTFjNzI1MTNjODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.yUdx6Agv2yyn7cl1hZhAXzi5VZwBuykux3F9bxutqS4">

Nice catch @kaaloo thanks !
Was tricky to fix ^^
